_posts/2020-02-24-transit-gateway.md

```diff
@@ -0,0 +1,263 @@
+
+![alt text](/img/VPC-CIDR-Subnet.png)
+
+We create a subnet for each availability zone to allow for high availability configuration of resources and services running inside a VPC.  In this case AWS Sydney Region (ap-southeast-2) has 3 availability zones.
+
+You are now ready to deploy your VPC and Subnet infrastructure to all your AWS accounts.
+
+### Transit Gateway Deployment
+With the base VPC's and subnets deployed, we can now start deploying Transit Gateway resources.
+#### Transit Gateway Cloudformation Snippet
+
+    Resources:
+        TransitGateway:
+            Type: "AWS::EC2::TransitGateway"
+            Properties:
+            AmazonSideAsn: 65200
+            Description: "Transit Gateway Networking between VPC and On-Prem VPN"
+            AutoAcceptSharedAttachments: "enable"
+            DefaultRouteTableAssociation: "disable"   
+            DefaultRouteTablePropagation: "disable" 
+            DnsSupport: "enable"
+            VpnEcmpSupport: "enable"
+            Tags:
+                - Key: "Name"
+                Value: "ACME Transit Gateway"
+    Outputs:
+        TransitGateway:
+            Description: Static Transit Gateway Id
+            Value: !Ref TransitGateway
+            Export:
+                Name: TransitGatewayId
+
+By disabling Default Route Table Association and Propagation, you get control over which Transit Gateway Table you can associate with, allowing you to restrict workload VPC(Eg, Sandox) to communicate with other Workload VPC(Eg, Production) or other specific requirements you may have.  AutoAcceptSharedAttachments are enabled so we don't have to manually accept invitations
+of shared resource for each account added to the AWS Organization Business Unit.  Choose a unique ASN number for AmazonSideAsn.  
+
+    TGWResourceAccessShare:
+        Type: AWS::RAM::ResourceShare
+        DependsOn: TransitGateway
+        Properties:
+        AllowExternalPrincipals: false
+        Name: "Transit Gateway Resource Share with ACME AWS Organization"
+        ResourceArns:
+            - !Sub "arn:aws:ec2:${AWS::Region}:${AWS::AccountId}:transit-gateway/${TransitGateway}"
+        Principals: 
+            - !Sub "arn:aws:organizations::${MasterAwsAccountId}:organization/${AwsACMEOrgId}"
+        Tags:
+            - Key: "Name"
+            Value: "ACME Organizarion TGW Resource Share"
+
+When sharing the Transit Gateway resource with the AWS Organization Id as the principal, all account members of the Organization you have created will automatically get access to the Transit Gateway resource in their VPC. Make sure you keep "AllowExternalPrincipals" as false, unless you really want to expose the Transit Gateway Resource to an AWS account you dont manage.
+
+### Site to Site VPN Deployment
```

Comment:
I may be wrong - but don't the site to site VPN's automatically close if they don't receive traffic for a specific time period?

https://aws.amazon.com/premiumsupport/knowledge-center/vpn-tunnel-troubleshooting/

Reply:
you may want to make folks aware of that... that might cause confusion

Reply:
good call.. it does timeout, if no keepalive messages are sent through from on-prem, but there is no real disruption to user traffic except for initial traffic lag, takes on average 2-5 seconds for tunnel to come back up..  Users wont notice unless they have latency sensitive applications on startup